src/onegov/org/views/reservation.py

```diff
@@ -651,48 +652,53 @@ def accept_reservation(self, request, text=None, notify=False):
-        q = ResourceRecipientCollection(request.session).query()
-        q = q.filter(ResourceRecipient.medium == 'email')
-        q = q.order_by(None).order_by(ResourceRecipient.address)
-        q = q.with_entities(
-            ResourceRecipient.address,
-            ResourceRecipient.content
-        )
-        recipients = [
-            r.address
-            for r in q if (
-                self.resource.hex in r.content['resources']
-                and r.content.get('new_reservations', False)
-            )
-        ]
-
-        # E-mail for new reservations
-        args = {
-            'layout': DefaultMailLayout(object(), request),
-            'title': request.translate(
-                _("${org} New Reservation(s)", mapping={
-                    'org': request.app.org.title
-                })
-            ),
-            'form': form,
-            'model': self,
-            'resource': resource,
-            'reservations': reservations,
-            'show_submission': show_submission,
-            'message': message
-        }
-
-        content = render_template(
-            'mail_new_reservation_notification.pt', request, args
-        )
-
-        for r in recipients:
-            request.app.send_transactional_email(
-                subject=args['title'],
-                receivers=(r),
-                content=content,
-            )
+        def recipients_which_have_registered_for_mail():
+            q = ResourceRecipientCollection(request.session).query()
+            q = q.filter(ResourceRecipient.medium == 'email')
+            q = q.order_by(None).order_by(ResourceRecipient.address)
+            q = q.with_entities(ResourceRecipient.address,
+                                ResourceRecipient.content)
+
+            for res in q:
+                if self.resource.hex in res.content['resources'] \
+                        and res.content.get('new_reservations', False):
+                    yield res.address
+
+        def email_iter():
+            for recipient_addr in recipients_which_have_registered_for_mail():
+                title = request.translate(
+                    _(
+                        "${org} New Reservation(s)",
+                        mapping={'org': request.app.org.title},
+                    )
+                )
+
+                content = render_template(
+                    'mail_new_reservation_notification.pt',
+                    request,
+                    {
+                        'layout': DefaultMailLayout(object(), request),
+                        'title': title,
+                        'form': form,
+                        'model': self,
+                        'resource': resource,
+                        'reservations': reservations,
+                        'show_submission': show_submission,
+                        'message': message,
+                    },
+                )
+                plaintext = html_to_text(content)
```

Comment:
You should move this outside the loop, the title,  content and plaintext are the same for all the emails.

Reply:
Yes (!) I can't believe I didn't see that.

Reply:
The two methods in `ticket.py` and `reservation.py` share a lot of characteristics (in fact, differ only in detail) it might be another improvement to pull the logic upwards and extract this into a shared method. I'm not sure if it's a good idea though, maybe a bit overkill?

Reply:
You could probably pull `mail_iter` out into a utility function for mail batches that have static content.

It would accept an `Iterable[str]` of recipients, as well as `subject`, `html`, `plaintext` where you only need to provide either `html` or `plaintext`, if only `html` is given, `plaintext` is created using `html2text`.